src/util/system.cpp

```diff
@@ -1062,9 +1062,14 @@ void ArgsManager::LogArgs() const
 
 bool RenameOver(fs::path src, fs::path dest)
 {
+#ifdef __MINGW64__
+    return MoveFileExW(src.wstring().c_str(), dest.wstring().c_str(),
```

Comment:
Might want to add a comment here specifying why this workaround exists, and we should probably file an upstream issue and refer to it, so it has a limited lifetime.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/24331#issuecomment-1039185324).